### PR TITLE
Added oscap stuff

### PR DESCRIPTION
**Pull request by kbidarkar**
Added most of the oscap pre-requesites related to policy, content
    and reports.
    
    *) Added OSCAP constant related to content, profile, period, e.t.c
    *) Added new function `select` to base module.
    *) Added `make_oscappolicy` to factory module.
    *) Refactored the ui/hostgroup module and added new code to
       populate content_source, puppet_ca and puppet_master.
    *) Added oscap and new hostgroup locators.
    *) Added oscap_policy, oscap_content, oscap_reports page
       navigation.

### CODE REVIEW

File: robottelo/vm.py [hunk 1]
```diff
@@ -272,6 +274,17 @@ def run(self, cmd):
...
+
+        """
+        domain = ssh.command(
+            u'hostname -d',
+            hostname=self.provisioning_server
+        )
+        return domain.stdout[0]
+
     def fetch_hostname(self):
         """Fetches short hostname from client and the domain from provisioning
```
Comment (kbidarkar) [inline robottelo/vm.py]: Plan is to remove this function and use vm.hostname , to avoid using `run` or `ssh.command`. Will remove this along with the test that uses this in my next PR.

File: robottelo/vm.py [hunk 2]
```diff
@@ -112,9 +112,11 @@ def create(self):
         if self.image_dir is not None:
             command_args.append('-p {image_dir}')
 
+        self.domain = self.fetch_domain()
+
         command = u' '.join(command_args).format(
             source_image=u'{0}-base'.format(self.distro),
-            target_image=self._target_image,
+            target_image=u'{0}.{1}'.format(self._target_image, self.domain),
```
Comment (kbidarkar) [inline robottelo/vm.py]: For puppet we need, client hostname to be set with  FQDN.

**Comment by kbidarkar** [comment]
will fix all the issues soon, had a bad typo while working with git and resolving conflicts, but now mostly all issue related to git resolved. Will raise another PR soon.

File: robottelo/vm.py (hunk 2 above)
Comment (kbidarkar) [inline robottelo/vm.py]: Also the vm targetname and hostname are tied up together.
```
 ~]# snap-guest --help | grep -B 2 -i "target" 
  --base [image] | -b [image]
        Base image name (template) - required
  --target [name] | -t [name]
        Target image name (and hostname) - required
--
```

**Comment by kbidarkar** [comment]
Git issues RESOLVED and flake8 has PASSED.

Current facing issues as vm unittest needs updation as the vm.create() functionality has been updated.
Working on it.

**Comment by kbidarkar** [comment]
This PR is now ready for review.

File: robottelo/vm.py [hunk 3]
```diff
@@ -293,6 +297,17 @@ def put(self, local_path, remote_path=None):
             )
         ssh.upload_file(local_path, remote_path, hostname=self.ip_addr)
 
+    def fetch_domain(self):
+        """Fetches domian name from the domain of the provisioning
+        server as it's bridge based network.
+
+        """
+        domain = ssh.command(
```
Comment (elyezer) [inline robottelo/vm.py]:
```python
return ssh.command(
    ...
).stdout[0]
```

File: robottelo/vm.py [hunk 4]
```diff
@@ -114,9 +115,12 @@ def create(self):
         if self.image_dir is not None:
             command_args.append('-p {image_dir}')
 
+        if self._domain is None:
+            self._domain = self.fetch_domain()
```
Comment (elyezer) [inline robottelo/vm.py]: fetch_domain could return an empty string if the provisioning server does not have a FQDN. We should raise `VirtualMachineError` if it happens.

File: tests/robottelo/test_vm.py [hunk 5]
```diff
@@ -30,8 +30,9 @@ def configure_provisoning_server(self):
 
     @patch('time.sleep')
     @patch('robottelo.ssh.command', side_effect=[
+        ssh.SSHCommandResult(stdout=[u'example.redhat.com']),
         ssh.SSHCommandResult(),
-        ssh.SSHCommandResult(stdout=['(192.168.0.1)'])
+        ssh.SSHCommandResult(stdout=['(192.168.0.1)']),
```
Comment (elyezer) [inline tests/robottelo/test_vm.py]: :+1:

File: robottelo/vm.py [hunk 6]
```diff
@@ -293,6 +297,17 @@ def put(self, local_path, remote_path=None):
             )
         ssh.upload_file(local_path, remote_path, hostname=self.ip_addr)
 
+    def fetch_domain(self):
```
Comment (elyezer) [inline robottelo/vm.py]: This method is not related to the VM, we can rename it for now to provisioning_server_hostname.

The next step should be receive the domain as configuration and if not provided fetch it from the provisioning server..

Also the provisining server domain is the same as:

```python
domain = self.provisioning_server.split('.', 1)[1]
```

File: robottelo/vm.py [hunk 7]
```diff
@@ -304,12 +319,10 @@ def fetch_hostname(self):
         # returned from virtual machine is short hostname. It is being done to
         # handle the scenario where in the vm/client might return FQDN for
         # a different provisioning server.
+        # TODO: remove this function and use self.hostname to avoid ssh and
```
Comment (elyezer) [inline robottelo/vm.py]: :+1:

File: robottelo/ui/oscappolicy.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        self.click(common_locators['submit'])
+        self.configure_entity([host_group], FILTER['policy_hgrp'])
+        self.click(common_locators['submit'])
+
+    def search(self, name):
+        """Searches existing oscap policy from UI"""
+        self.wait_for_ajax()
+        Navigator(self.browser).go_to_oscap_policy()
+        strategy, value = common_locators['search']
+        element = self.wait_until_element((strategy, value % name))
```
Comment (elyezer) [inline robottelo/ui/oscappolicy.py]: `return self.wait_until_element((strategy, value % name))`

File: robottelo/ui/oscapcontent.py [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+            raise UIError(u'Could not create new content {0}'.format(name))
+        self.text_field_update(locators['oscap.content_title'], name)
+        # TODO: upload oscap content
+        self.click(common_locators['submit'])
+
+    def search(self, name):
+        """Searches existing oscap content from UI"""
+        self.wait_for_ajax()
+        Navigator(self.browser).go_to_oscap_content()
+        element = self.search_entity(
```
Comment (elyezer) [inline robottelo/ui/oscapcontent.py]: `return self.search_entity(...`

File: robottelo/ui/factory.py [hunk 10]
```diff
@@ -601,8 +605,26 @@ def make_ldapauth(session, **kwargs):
         u'account_basedn': None,
         u'account_grpbasedn': None,
         u'ldap_filter': False,
-        u'otf_register': True
+        u'otf_register': True,
```
Comment (elyezer) [inline robottelo/ui/factory.py]: :+1:

File: robottelo/ui/base.py [hunk 11]
```diff
@@ -471,3 +473,25 @@ def click(self, locator, wait_for_ajax=True,
...
+            clicking on the element
+        :param timeout: The amount of time that wait_fox_ajax should wait. This
+            will have effect if ``wait_fox_ajax`` parameter is ``True``.
+        :raise: UINoSuchElementError if the element could not be found.
+
+        """
+        element = self.wait_until_element(locator)
+        if element is None:
+            raise UINoSuchElementError(
+                '{}: element with locator {} not found while trying to click.'
```
Comment (elyezer) [inline robottelo/ui/base.py]: s/click/select

**Comment by elyezer** [comment]
ACK, pending comments.

File: robottelo/ui/base.py (hunk 11 above)
Comment (kbidarkar) [inline robottelo/ui/base.py]: ah!, I had many issues while re-factoring this, had to drop and add code back , may be in that process missed it. :)

File: robottelo/vm.py (hunk 6 above)
Comment (kbidarkar) [inline robottelo/vm.py]: Totally agree, I think we better avoid using `run` or `ssh.command` if possible.

**Comment by Ichimonji10** [comment]
@kbidarkar Do you have any test results?

ACK pending question.

**Comment by kbidarkar** [comment]
I had been working on this since long time, so thought let me split it up in 2, 
a) oscap_pre-requisites (crud functions, locators, navigators, e.t.c)
b) oscap_tests

In my next PR will have some test results :)

**Comment by kbidarkar** [comment]
Also, the test-cases needs more clean up and optimization so me on it.

**Comment by Ichimonji10** [comment]
Ahh, so this is the first PR in a pair of PRs, where this one adds tools necessary for actually writing tests? Sounds sane to me. ACK.
